ocaml-lsp-server/docs/ocamllsp/typeEnclosing-spec.md

```diff
@@ -0,0 +1,51 @@
+# Type Enclosing Request
+
+## Description
+
+Merlin has a concept of `type enclosing` that gets the type of ident under the
+cursor. It will highlight the ident and display its type. You can climb the
+typed-tree and display the type of bigger expressions surrounding the cursor.
+Since _LSP_ is stateless, the manipulation related to growing or shrinking
+enclosings is delayed to the client. This request allows to request type
+enclosing under the cursor with a support of parent enclosings.
+
+## Client capability
+
+There is no client capability relative to this request.
+
+## Server capability
+
+- property name: `handleTypeEnclosing`
+- property type: `boolean`
+
+## Request
+
+- method: `ocamllsp/typeEnclosing`
+- params:
+
+  ```json
+  {
+    "textDocument": TextDocumentIdentifier,
+    "position": Position,
+    "workDoneToken?": ProgressToken,
+    "index": uinteger,
+    "verbosity": uinteger,
+    "rangeEnd?": Position
+  }
```

Comment:
I think I forgot the `uri` of the document in my specification, we should respect this interface:
https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#textDocumentPositionParams

Not sure about the utility of the `workDoneToken` however.

Reply:
I think it is already handle by `textDocumentPosition` (that has a `texDocument` field, which is a `textDocumentIdentifier`, that has a `uri` field :) )

Reply:
Oh sorry ! I didn't see that :-)